bbc/simorgh
# Refactor list item transformation logic in OJ container components

Resolves #7289

**Overall change:**
Simplify the OJ container components by moving the logic that renders the story promos into newly created Promo and PromoList components

**Code changes:**

- For CpsRecommendations and CpsRelatedContent created one Promo component using the `singleTransform` function and one PromoList component using the `listTransform` function
- Removed `singleTransform` and `listTransform` from CpsFeaturesAnalysis and CpsTopStories,

- In CpsOnwardJourney container:
  - Renamed `singleTransform` as `promoComponent`
  - Renamed `listTransform` as `promoListComponent`
  - Passed in the new RecommendationsPromo, RecommendationsPromoList, RelatedContentPromo and RelatedContentPromoList components
---


**Testing:**


- Check for regressions on STY page: http://localhost:7080/mundo/23263889 for the following Onward Journey (OJ) components: 
  - Recommendations (Quizás también te interese)
  - RelatedContent (Contenido relacionado)
  - TopStories (Principales noticias) 
  - FeaturesAnalysis (No te lo pierdas)

- Check for regressions on PGL page: http://localhost:7080/mundo/deportes-36935058 for RelatedContent (Contenido relacionado)

- Check for regressions on MAP page: http://localhost:7080/pidgin/23248703 for RelatedContent (Another thing we de for inside dis tori)
